src/App.vue

```diff
@@ -90,26 +111,126 @@ export default class App extends Mixins(DateMixin, FilingTemplateMixin, LegalApi
   @Action setCurrentDate!: ActionBindingIF
   @Action setCertifyStatementResource!: ActionBindingIF
   @Action setNameRequestState!: ActionBindingIF
+  @Action setAuthRoles: ActionBindingIF
 
   // Local Properties
   private filingData: Array<FilingDataIF> = []
   private draftFiling: IncorporationFilingIF
+  private isDraft: boolean = false
+  private accountAuthorizationErrorDialog: boolean = false
+  private nameRequestInvalidErrorDialog: boolean = false
+  private nameRequestInvalidType: string = ''
+  private evaluatingPreconditions: boolean = true
 
   private async created (): Promise<any> {
-    // Mock the nrNumber and Data:
-    this.setNameRequestState({ nrNumber: 'NR7654560', entityType: 'BC' })
+    // Mock the nrNumber and Data
+    const nameRequest = await this.evaluateNRPreconditions()
+    if (nameRequest && nameRequest.nrNum && nameRequest.isConsumable) {
+      // TODO: Handling different NR Formats
```

Comment:
Is it now possible to handle BCOMP vs COOPS NRs? If so then please update this code accordingly.

Reply:
Unsure of what data from the NR response I would use to determine this, going to take a look at the API to see what types would tie to BCOMP or COOPs for us.

Reply:
Taking a look at https://github.com/bcgov/namex/blob/master/api/namex/constants/__init__.py#L22 , the requestTypeCd attribute is probably what we are needing. I committed a change to set that in the name request state, but it looks like the NRs I have are of type CR (BC_CORPORATION).

Reply:
Yes, that looks right... class BCProtectedNameEntityTypes right?

Yes, the NRs Lorna gave us are all BC Corp. Apparently Benefit Companies don't exist yet in NRO or Namex.

For MVP we need to handle Benefit Companies, and later, Cooperatives. Do what you can with the data we have and we'll need to remember to review this later. Could you shoot Kaine a text/email to update him and see if he knows when we will Benefit Comps?

Reply:
Done, I gave Kaine an update on where we are at with this and he is looking into it.

Reply:
Update (should be by the end of the month):
```
we will have an attribute called entity_type_cd=BC (Benefit company)
also, you will know it is an incorporation by the request-action_cd='NEW' and entity_type='BC'
```